aten/src/ATen/cuda/CUDAGraph.h

```diff
@@ -18,22 +18,32 @@ namespace cuda {
 // to CUDAGraph::capture_begin
 TORCH_CUDA_CPP_API MempoolId_t graph_pool_handle();
 
-void dynamic_graph_updater(cudaGraphKernelNodeUpdate* updates, size_t num_updates);
-// Large parameter support:
-// https://developer.nvidia.com/blog/cuda-12-1-supports-large-kernel-parameters/
-// Question: Does pytorch still run on Pascal or older?
 
 struct KernelUpdateSOA {
-  static constexpr size_t MAX_NUM_UPDATES =
-      1364; // (32764 - sizeof(size_t)) / (sizeof(void*) + sizeof(size_t) +
-            // sizeof(cudaGraphDeviceNode_t));
+  // Even on Cuda 12.4, Torch supports compute capability down to sm_50
+  // However, large parameter support was only added in sm_70
+  // https://developer.nvidia.com/blog/cuda-12-1-supports-large-kernel-parameters/
+#if (__CUDA_ARCH__ >= 700)
+  static constexpr size_t KERNEL_PARAM_LIMIT_BYTES = 32764;
+#else
+  static constexpr size_t KERNEL_PARAM_LIMIT_BYTES = 4096;
+#endif
+  static constexpr size_t PER_UPDATE = sizeof(void*) + sizeof(size_t) + sizeof(cudaGraphDeviceNode_t);
+
+  static constexpr size_t MAX_NUM_UPDATES = (KERNEL_PARAM_LIMIT_BYTES - sizeof(size_t)) / PER_UPDATE;
+
   size_t num_updates;
+  // should this a struct? like: KernelUpdate updates[MAX_NUM_UPDATES];
```

Comment:
No. I want a struct of arrays (thus, "SOA") to maximize the number of coalesced global memory accesses.

Reply:
Gotcha!